From 5aac7a48924babe86b64429585375f5c49a31850 Mon Sep 17 00:00:00 2001
From: Linu Cherian <lcherian@marvell.com>
Date: Thu, 27 Sep 2018 11:32:39 +0530
Subject: [PATCH 0212/1051] soc: octeontx2: Skip cgx linkup start for invalid
 ports.

Skip cgx linkup start procedure for invalid ports.

Fixes: df5bd7baa793 ("soc: octeontx2: Make cgx link up procedure threaded during boot")
Signed-off-by: Linu Cherian <lcherian@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2/cgx.c     | 3 +++
 drivers/soc/marvell/octeontx2/rvu_cgx.c | 2 ++
 2 files changed, 5 insertions(+)

diff --git a/drivers/soc/marvell/octeontx2/cgx.c b/drivers/soc/marvell/octeontx2/cgx.c
index b521df10411d..d3a4cb711be6 100644
--- a/drivers/soc/marvell/octeontx2/cgx.c
+++ b/drivers/soc/marvell/octeontx2/cgx.c
@@ -642,6 +642,9 @@ int cgx_lmac_linkup_start(void *cgxd)
 	struct cgx *cgx = cgxd;
 	struct task_struct *task;
 
+	if (!cgx)
+		return -ENODEV;
+
 	/* Start the linkup procedure of lmac ports in the background */
 	task = kthread_run(cgx_lmac_linkup_thread, cgx, "cgx%d_linkup_thread",
 			      cgx->cgx_id);
diff --git a/drivers/soc/marvell/octeontx2/rvu_cgx.c b/drivers/soc/marvell/octeontx2/rvu_cgx.c
index 727889a7ff1f..a1c7c23a5b35 100644
--- a/drivers/soc/marvell/octeontx2/rvu_cgx.c
+++ b/drivers/soc/marvell/octeontx2/rvu_cgx.c
@@ -235,6 +235,8 @@ static void cgx_lmac_event_handler_init(struct rvu *rvu)
 
 	for (cgx = 0; cgx <= rvu->cgx_cnt_max; cgx++) {
 		cgxd = rvu_cgx_pdata(cgx, rvu);
+		if (!cgxd)
+			continue;
 		for (lmac = 0; lmac < cgx_get_lmac_cnt(cgxd); lmac++) {
 			err = cgx_lmac_evh_register(&cb, cgxd, lmac);
 			if (err)
-- 
2.17.1

